src/ADAL.PCL/Internal/Telemetry/HttpEvent.cs

```diff
@@ -0,0 +1,124 @@
+{
+    internal class HttpEvent : DefaultEvent
+    {
+        internal string UserAgent { get; set; }
+
+        internal string HttpMethod { get; set; }
+
+        internal HttpStatusCode HttpResponseCode { get; set; }
+
+        internal string HttpQueryParameters { get; set; }
+
+        internal string HttpResponseMethod { get; set; }
+
+        internal string ResponseApiVersion { get; set; }
+
+        internal void ParseQuery(string Query)
+        {
+            if (Query.Length != 0)
+            {
+                string QueryShifted = Query.Substring(1, Query.Length - 1);
+                string[] result = QueryShifted.Split('&');
+                StringBuilder stringBuilder = new StringBuilder();
+                foreach (string s in result)
+                {
+                    if (s.Contains("="))
+                    {
+                        stringBuilder.Append(s.Split('=')[0]).Append("&");
+                    }
+                }
+
+                if (string.IsNullOrEmpty(stringBuilder.ToString()))
+                {
+                    SetEvent(EventConstants.ExtraQueryParameters,
+                        stringBuilder.ToString().Substring(0, stringBuilder.Length - 1));
+                }
+            }
+        }
+
+        internal override void ProcessEvent(Dictionary<string, string> dispatchMap)
+        {
+            if (dispatchMap.ContainsKey(EventConstants.HttpEventCount))
+            {
+                int previousCount = int.Parse(dispatchMap[EventConstants.HttpEventCount]);
+                dispatchMap[EventConstants.HttpEventCount] = previousCount++.ToString();
+            }
+            else
+            {
+                dispatchMap.Add(EventConstants.HttpEventCount, "1");
+            }
+
+            if (dispatchMap.ContainsKey(EventConstants.HttpStatusCode))
+            {
+                dispatchMap[EventConstants.HttpStatusCode] = string.Empty;
+            }
+
+            if (dispatchMap.ContainsKey(EventConstants.RequestIdHeader))
+            {
+                dispatchMap[EventConstants.RequestIdHeader] = string.Empty;
+            }
+
+            if (dispatchMap.ContainsKey(EventConstants.OauthErrorCode))
+            {
+                dispatchMap[EventConstants.OauthErrorCode] = string.Empty;
+            }
+
+            if (dispatchMap.ContainsKey(EventConstants.HttpPath))
+            {
+                dispatchMap[EventConstants.HttpPath] = string.Empty;
+            }
+
+            foreach (KeyValuePair<string, string> Event in EventDictitionary)
+            {
+                if (Event.Key.Equals(EventConstants.HttpStatusCode) ||
+                    Event.Key.Equals(EventConstants.RequestIdHeader) ||
+                    Event.Key.Equals(EventConstants.OauthErrorCode) ||
+                    Event.Key.Equals(EventConstants.HttpPath)
+                    )
+                {
+                    if (dispatchMap.ContainsKey(Event.Key))
```

Comment:
resolved. http://stackoverflow.com/questions/4245064/method-to-add-new-or-update-existing-item-in-dictionary